### PR TITLE
Allow for offline configuration of CarbonManager

### PR DESCRIPTION
**Pull request by vanitasvitae**
As discussed earlier, it would make sense to allow configuration of the CarbonManager while the connection is not yet established.

This PR adds such functionality by introducing the variable `shouldBeEnabled`, which is checked in the managers `ConnectionListener.onAuthenticated()` block. If true, the manager will automatically try to enable carbons for the connection. Exceptions are logged and the user is still able to check if carbons are successfully enabled using the old `isEnabled()` methods.

### CODE REVIEW

**Comment by vanitasvitae** [comment]
I tried to make the changes as less intrusive as possible. Someone who doesn't follow Smacks development should not notice any differences in behaviour, apart from that they only have to enable Carbons once ;)

File: smack-experimental/src/main/java/org/jivesoftware/smackx/carbons/CarbonManager.java [hunk 1]
```diff
@@ -99,19 +105,30 @@ public void connectionCreated(XMPPConnection connection) {
     private final Set<CarbonCopyReceivedListener> listeners = new CopyOnWriteArraySet<>();
 
     private volatile boolean enabled_state = false;
+    private volatile boolean shouldBeEnabled = SHOULD_BE_ENABLED_DEFAULT;
```
Comment (Flowdalic) [inline smack-experimental/src/main/java/org/jivesoftware/smackx/carbons/CarbonManager.java:108]: Can we drop the "should be" everywhere? I'd prefer "enabled by default".

File: smack-experimental/src/main/java/org/jivesoftware/smackx/carbons/CarbonManager.java [hunk 2]
```diff
@@ -239,12 +263,10 @@ public boolean isSupportedByServer() throws NoResponseException, XMPPErrorExcept
      * You should first check for support using isSupportedByServer().
      *
      * @param new_state whether carbons should be enabled or disabled
-     * @throws NotConnectedException if the XMPP connection is not connected.
-     * @throws InterruptedException if the calling thread was interrupted.
      * @deprecated use {@link #enableCarbonsAsync(ExceptionCallback)} or {@link #disableCarbonsAsync(ExceptionCallback)} instead.
      */
     @Deprecated
-    public void sendCarbonsEnabled(final boolean new_state) throws NotConnectedException, InterruptedException {
+    public void sendCarbonsEnabled(final boolean new_state) {
```
Comment (Flowdalic) [inline smack-experimental/src/main/java/org/jivesoftware/smackx/carbons/CarbonManager.java:269]: nice catch!

**Comment by Flowdalic** [comment]
Squash into one commit please.

**Comment by vanitasvitae** [comment]
Rebase in progress :)

**Pull request merged**